libpod/oci_conmon_linux.go

```diff
@@ -1274,15 +1280,26 @@ func prepareProcessExec(c *Container, cmd, env []string, tty bool, cwd, user, se
 		pspec.User = processUser
 	}
 
-	hasHomeSet := false
-	for _, s := range pspec.Env {
-		if strings.HasPrefix(s, "HOME=") {
-			hasHomeSet = true
-			break
-		}
+	// We need to start with a fresh environment and should not inherit from the
+	// container's config (https://github.com/containers/libpod/issues/5255).
+	imageData, err := getImageDataFromContainer(c)
```

Comment:
Do we need this? We have the full list of environment variables the container was created from in the OCI spec.

Reply:
According to #5255 we need that to match Docker which always starts with the default + from the image.

Reply:
We don't want to use the extra environment variables set by the container process, but we do want those set at container creation time.

Tested with:
Terminal 1 - `sudo docker run -t -i --rm --name test --env a=b fedora bash`
Terminal 2 - `sudo docker exec -t -i test env`

I see `a=b` in the output of exec - so we should be including everything from container create time, but nothing that was added after the container was run.